### PR TITLE
Feat/replacing wagmi

### LINKED ISSUE #261: [Frontend] Replace wagmi with ethers for contrat write and read

### PR DESCRIPTION
**Pull request by Malak67**
# Generic Pull Request template

## Linked issue

Please link the issue:

ex. Closes #261 .

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.
- [x] The changes impact front-end.

## For changes impacting frontend

Please leave some screenshots/video recordings of the newly implemented interface and the Figma link for its initial design.

### Screenshots/ recordings

=> Multimedia here.

### Figma link

=> Figma link here.

## Description

Please provide a short description about what the PR contains.

## Funny gif

Leave a funny gif for faster review.

### CODE REVIEW

**Review by VargaElod23 — DISMISSED** [review]
LGTM 👍🏼

File: services/client/src/hooks/useEscrow.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      try {
+        const tx: ethers.providers.TransactionResponse = await escrowContract!.claim(
+          args.receiver,
+          args.poolId,
+          args.amount,
+          args.tokenAddress,
+          args.nonce,
+          args.hash,
+          {
+            gasLimit: BigNumber.from(9999999),
```
Comment (rattrap) [inline services/client/src/hooks/useEscrow.tsx:57]: We can remove this.

File: services/client/src/hooks/useEscrow.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+      showLoading(['Please, sign the message...', 'Funding the pool...']);
+      try {
+        const tx: ethers.providers.TransactionResponse = await escrowContract!.deposit(
+          spender,
+          poolId,
+          amount,
+          tokenAddress,
+          {
+            from: spender,
+            gasLimit: BigNumber.from(9999999),
```
Comment (rattrap) [inline services/client/src/hooks/useEscrow.tsx:89]: Remove fixed gasLimit

File: services/client/src/hooks/useEscrow.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            value: amount,
```
Comment (rattrap) [inline services/client/src/hooks/useEscrow.tsx:90]: If this is an ERC20 token, we don't need to set the value

File: services/client/src/hooks/useEscrow.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        await tx.wait();
+        hideLoadingModal();
+        return await deposit(spender as AddressType, poolId, amount, tokenAddress);
+      } catch (error: any) {
+        console.log('On error: ', error);
+        hideLoadingModal();
+        showNotificationModal(NotificationType.ERROR, error?.message);
+      }
+    },
+    [provider, deposit, erc20ABI, hideLoadingModal, showLoading, showNotificationModal],
```
Comment (rattrap) [inline services/client/src/hooks/useEscrow.tsx:128]: Why does this depend on the deposit callback?

File: services/client/src/hooks/useEscrow.tsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        hideLoadingModal();
+        showNotificationModal(NotificationType.ERROR, error?.message);
+      }
+    },
+    [escrowContract, hideLoadingModal, showLoading, showNotificationModal],
+  );
+
+  const approve = useCallback(
+    async (
+      spender: string,
```
Comment (rattrap) [inline services/client/src/hooks/useEscrow.tsx:107]: I think the spender is always our Escrow contract

File: services/client/src/hooks/useHypePools.tsx [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+import { HypePool } from '../models';
+import { useLoadingModals } from './useLoadingModals';
+import { NotificationType } from '../utils';
+import ABIs from '../abi';
+
+export interface WritePoolDetailsArgs {
+  title: string;
+  projectName: string;
+  tokenName?: string;
+  word: string;
```
Comment (rattrap) [inline services/client/src/hooks/useHypePools.tsx:13]: Don't forget to rename this

File: services/client/src/pages/AddHypePool/Summary/Summary.effects.tsx [hunk 7]
```diff
@@ -97,17 +54,21 @@ export const useSummaryEffects = (
         );
       } else {
         if (isCustomToken) {
-          setEnableApprove(true);
+          await approve(account, createdPoolIndex, amount, rewards.tokenAddress as AddressType);
```
Comment (rattrap) [inline services/client/src/pages/AddHypePool/Summary/Summary.effects.tsx:57]: We still need to call the deposit function even if it's a custom token

File: services/client/src/pages/PoolDetails/PoolDetails.effects.tsx [hunk 8]
```diff
@@ -116,17 +76,19 @@ export const usePoolDetailsEffects = (poolId: string) => {
         );
       } else {
         if (isCustomToken) {
-          setEnableApprove(true);
+          await approve(account, poolId, amount, pool?.tokenAddress as AddressType);
```
Comment (rattrap) [inline services/client/src/pages/PoolDetails/PoolDetails.effects.tsx:79]: Same. Still need to call deposit after approval

**Review by rattrap — CHANGES_REQUESTED** [review]

**Review by rattrap — APPROVED** [review]

**Review by VargaElod23 — APPROVED** [review]

**Pull request merged**